pallets/msa/src/types.rs

```diff
@@ -175,12 +175,52 @@ impl AddProvider {
 	/// Create new `AddProvider`
 	pub fn new(
 		authorized_msa_id: MessageSourceId,
-		schema_ids: Option<Vec<SchemaId>>,
+		intent_ids: Option<Vec<IntentId>>,
 		expiration: BlockNumber,
 	) -> Self {
-		let schema_ids = schema_ids.unwrap_or_default();
+		let intent_ids = intent_ids.unwrap_or_default();
 
-		Self { authorized_msa_id, schema_ids, expiration }
+		Self { authorized_msa_id, intent_ids, expiration }
+	}
+}
+
+/// Structure that is signed for granting permissions to a Provider
+#[derive(TypeInfo, Clone, Debug, Decode, DecodeWithMemTracking, Encode, PartialEq, Eq)]
+pub struct AddProviderV2 {
+	/// The provider being granted permissions
+	pub authorized_msa_id: MessageSourceId,
+	/// Schemas for which publishing grants are authorized.
+	/// This is private intended for internal use only.
+	pub intent_ids: Vec<IntentId>,
+	/// The block number at which the proof for grant_delegation expires.
+	pub expiration: BlockNumber,
+}
+
+impl EIP712Encode for AddProviderV2 {
+	fn encode_eip_712(&self, chain_id: u32) -> Box<[u8]> {
+		lazy_static! {
+			// signed payload
+			static ref MAIN_TYPE_HASH: [u8; 32] = sp_io::hashing::keccak_256(
+				b"AddProvider(uint64 authorizedMsaId,uint16[] intentIds,uint32 expiration)"
```

Comment:
```suggestion
				b"AddProviderV2(uint64 authorizedMsaId,uint16[] intentIds,uint32 expiration)"
```

I think in other places we tried to use the same struct name